### PR TITLE
add delete proc for JArray, remove by index, no check here,use system…

### LINKED ISSUE #15001: JSON delete item from JArray
### Summary

- Add 1 convenience proc to remove an item from a `JArray`.

```nim
import json
var jason = parseJson("""["x", "y"]""")
jason.delete "y"  # OR .pop()  ?
# ["x"]
```

Can be by index or by value or both, maybe like a template using `json.to(...,  seq[string])` or `json.getElems()`, etc.
:)

### PR DESCRIPTION
**Pull request by bung87**
….delete defualt behavior

Fix https://github.com/nim-lang/Nim/issues/15001

### CODE REVIEW

**Comment by juancarlospaco** [comment]
Needs changelog, since pragma, maybe `{.inline.}`.

File: lib/pure/json.nim [hunk 1]
```diff
@@ -566,6 +566,11 @@ proc delete*(obj: JsonNode, key: string) =
     raise newException(KeyError, "key not in object")
   obj.fields.del(key)
 
+proc delete*(obj: JsonNode, i: Natural) =
```
Comment (ee7) [inline lib/pure/json.nim]:
```suggestion
proc delete*(obj: JsonNode, i: Natural) {.since: (1, 5).} =
```

File: lib/pure/json.nim [hunk 2]
```diff
@@ -566,6 +566,11 @@ proc delete*(obj: JsonNode, key: string) =
...
+  ## Deletes ``arr[i]``.
```
Comment (ee7) [inline lib/pure/json.nim]:
```suggestion
  ## Deletes ``obj[i]``.
```

This also makes it more consistent with the existing [json.delete](https://github.com/nim-lang/Nim/blob/a81434a890d70e7146c440c59caadc701c6099b3/lib/pure/json.nim#L562..L563)
```Nim
proc delete*(obj: JsonNode, key: string) =
  ## Deletes ``obj[key]``.
```

**Review by ee7 — COMMENTED** [review]
Looks like this PR is essentially Araq's suggestion for the second proc in PR https://github.com/nim-lang/Nim/pull/15790

**Comment by bung87** [comment]
> Looks like this PR is essentially Araq's suggestion for the second proc in PR #15790

yeah,I first find issue when I done this, i found that PR, but that would not accept, it has unnessery bound check,  system.delete does not raise too, also in js ,when index greater than hight,it just remove high,also delete by value may not usefull ,there may multiple value equals

**Comment by timotheecour** [comment]
> also delete by value may not usefull ,there may multiple value equals [...]

see my proposal https://github.com/nim-lang/Nim/pull/15790#issuecomment-783102954 instead

> system.delete does not raise too, also in js ,when index greater than hight,it just remove high

see https://github.com/nim-lang/Nim/issues/16544

**Pull request closed without merging**